trl/trainer/ppov2_bandit_rloo_trainer.py

```diff
@@ -0,0 +1,704 @@
+        self.policy.generation_config.pad_token_id = None  # generate tokens without truncation / padding
+
+        self.ref_policy = ref_policy
+        self.reward_model = reward_model
+        self.train_dataset = train_dataset
+        self.train_dataset_len = len(train_dataset)
+        self.train_generation_config = train_generation_config
+        self.data_collator = data_collator
+        self.eval_dataset = eval_dataset
+        self.eval_generation_config = eval_generation_config
+        if eval_generation_config is None:
+            self.eval_generation_config = train_generation_config
+        self.optimizer, self.lr_scheduler = optimizers
+        self.callbacks = callbacks
+
+        #########
+        # calculate various batch sizes
+        #########
+        accelerator = Accelerator(gradient_accumulation_steps=args.gradient_accumulation_steps)
+        self.accelerator = accelerator
+        args.world_size = accelerator.num_processes
+        args.local_batch_size = args.per_device_train_batch_size * args.gradient_accumulation_steps * args.nminibatches
+        args.micro_batch_size = int(args.per_device_train_batch_size * args.world_size)
+        args.batch_size = int(args.local_batch_size * args.world_size)
+        args.mini_batch_size = exact_div(args.batch_size, args.nminibatches)
+        args.local_mini_batch_size = exact_div(args.local_batch_size, args.nminibatches)
+        if args.whiten_rewards:
+            assert (
+                args.local_mini_batch_size >= 8
+            ), f"Per-rank minibatch size {args.local_mini_batch_size} is insufficient for whitening"
+        # `per_rank_rollout_batch_size` is our `args.local_batch_size`
+        # `per_rank_minibatch_size` is our `args.local_mini_batch_size`
+        args.num_updates = args.total_episodes // args.batch_size
+        time_tensor = torch.tensor(int(time.time()), device=accelerator.device)
+        time_int = broadcast(time_tensor, 0).item()  # avoid different timestamps across processes
+        args.run_name = f"{args.exp_name}__{args.seed}__{time_int}"
+        self.local_seed = args.seed + accelerator.process_index * 100003  # Prime
+
+        #########
+        # disable dropout
+        #########
+        for module in [policy, ref_policy, reward_model]:
+            disable_dropout(module)
+        policy.generation_config.eos_token_id = (
+            None  # disable `pad_token_id` and `eos_token_id` because we just want to
+        )
+        policy.generation_config.pad_token_id = None  # generate tokens without truncation / padding
+
+        #########
+        # setup model, optimizer, and others
+        #########
+        if args.truncate_token and args.truncate_token == "eos":
+            args.truncate_token_id = tokenizer.eos_token_id
+        self.model = policy
+        self.create_optimizer_and_scheduler(num_training_steps=args.num_updates)
```

Comment:
Should `num_training_steps` be a function of `self.train_dataset` and the number of epochs and batch size, so it's automatic and the user doesn't need to calculate this?

Reply:
TODO: I will think about this.

Reply:
I added this:

```
        if args.total_episodes is None:  # allow the users to define episodes in terms of epochs.
            args.total_episodes = args.num_train_epochs * self.train_dataset_len
```